source/assets/js/components/fef-selectable.js

```diff
@@ -0,0 +1,289 @@
+            let savedSelections = FefStorage.getItemJsonParsed(STORAGE_KEY);
+
+            let selectedSourceURN = savedSelections[this.collectionURN];
+
+            if (selectedSourceURN && this.connectsToCollection(selectedSourceURN)) {
+                this.$element.hide();
+                this.toggleCollections(selectedSourceURN);
+
+                let $collection = $(this.$sourceCollections.toArray().find(c => $(c).data('urn') === selectedSourceURN));
+
+                if ($collection) {
+                    $collection.addClass(SELECTED_COLLECTION_CLASS);
+                    return;
+                }
+            }
+        }
+
+        // none was selected: hide all collections, show selection element
+        this.toggleCollections(false);
+        this.$element.show();
+    }
+
+    /**
+     * Upon changing a new 'Option', the following things happen:
+     * - collection representing the selected landingpage is shown, the others hidden
+     * - empty state (this.$element) gets hidden
+     * - selection is saved in localstorage
+     * - tracking
+     *
+     * @param {string} sourceURN
+     * @param {jQuery.event} e
+     */
+    changeSelectedSource(sourceURN, e) {
+        let nextSource = this.sources.find(source => source.urn === sourceURN);
+
+        if (!nextSource) {
+            throw new Error(`Source '${ sourceURN }' can't be found in the selectable collection ${ this.collectionURN }.`);
+        }
+
+        this.showSourceCollection(sourceURN, e);
+        this.saveSelectionInLocalStorage(sourceURN);
+
+        this.track(this.selectedSource, nextSource);
+        this.selectedSource = nextSource;
+    }
+
+    /**
+     * Immediately shows/hides collections, depending on their urn.
+     *
+     * @param {string} sourceURN URN of a collection that acts as a source
+     */
+    toggleCollections(sourceURN) {
+        this.$sourceCollections.each((_, collection) => {
+            let $collection = $(collection),
+                willBeShown = $collection.data('urn') === sourceURN;
+            $collection.toggle(willBeShown);
+        });
+    }
+
+    /**
+     * Shows the selection element
+     * @param {jQuery.event} event Original event that triggered the change
+     */
+    showSelectionElement(event) {
+        let $collection = this.$sourceCollections.filter(`.${SELECTED_COLLECTION_CLASS}`).first(),
+            shouldFocus = event.screenX == 0 && event.screenY == 0;
+
+        if (!$collection) {
+            this.$element.show();
+            return;
+        }
+
+        this.$element.css({'position': 'absolute', 'height': 0}).show();
+        let $contentWrapper = $collection.find('.js-collection-content-wrapper');
+        let newHeight = this.$animationWrapper.height();
+        this.$animationWrapper.css('opacity', 0);
+
+        if (shouldFocus) {
+            $(':focus').blur();
+        }
+
+        $contentWrapper.animate({'opacity': 0}, ANIMATION_PART_DURATION, () => {
```

Comment:
und das lässt sich nicht mit css lösen, nehme ich an (?)

Reply:
Doch schon :) Aber Ich möchte nicht 7 Klassen machen müssen für jeden Zwischenstand der Animation, auf 4 Elemente verteilt. Macht's recht unwartbar in meinen Augen.

Reply:
👍